src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/FSharpCoreReflectionProxy.cs

```diff
@@ -0,0 +1,224 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Reflection;
+
+namespace System.Text.Json.Serialization.Metadata
+{
+    // Recognizing types emitted by the F# compiler requires consuming APIs from the FSharp.Core runtime library.
+    // Every F# application ships with a copy of FSharp.Core, however it is not available statically to System.Text.Json.
+    // The following class uses reflection to access the relevant APIs required to detect the various F# types we are looking to support.
+
+    /// <summary>
+    /// The various categories of F# types that System.Text.Json supports.
+    /// </summary>
+    internal enum FSharpKind
+    {
+        Unrecognized, Option, List, Set, Map, Record
```

Comment:
(nit), we usually put these one per line in C#

Reply:
Thanks, addressed your feedback in https://github.com/dotnet/runtime/pull/55108/commits/1a3662adf854b136027295b5b1f7e8a53c02c631.